src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -921,6 +929,8 @@ bool async_manifest_view::in_stm(async_view_search_query_t o) {
                    >= _stm_manifest.get_start_kafka_offset().value_or(
                      kafka::offset::min());
 
+          // The last timestamp in the archive is used as a pivot point. See
```

Comment:
I think it's not problematic since when start_kafka_offset == last_kafka_offset == nullopt, we'll not overlap from L927, but just want to make sure, should L930 be kafka::offset::max()? It reads like a bug, but I don't think it matters in practice

Reply:
This code is pretty and looks funny to me even after this change. Updated it though with your suggestion.

Reply:
It is funny because some methods return an optional, which still can be model::offset{} so they have 2 ways to communicate "unset". Further, the API doesn't communicate any relation between e.g. start being set and end being unset.

Other methods, like `get_archive_start_kafka_offset` will return model::offset{} for empty archive. Not an optional.